### ISSUE TITLE
Default behavior for restart

### ISSUE THREAD

**Issue by aannabe** [QMCPACK/qmcpack#2011]
If one decides that the error bar of a DMC run is not small enough, a restart/continuation might be attempted as:
```
<mcwalkerset fileroot="gs_dmc.s001"  version="3 8" collected="yes"/>
<qmc method="dmc" move="pbyp" checkpoint="1" append="yes">
```
However, this overwrites the scalar file of that series. I think this behavior is risky and appending the \*.scalar.\* , \*.dmc.\* , \*.stat.h5 might be better. In fact, it would be very intuitive if re-running the same original input would check for existing walkers and resume the calculations by appending all the mentioned files (this is of course a matter of preference).
In short, it's not completely clear from the documentation how to reduce the error bar of a particular series N.

**Comment by ye-luo** [comment]
I'm more concerned of corrupted files if a run was cut by walltime. Scanning for the latest sequence is not really possible. People can put gaps in the sequence, 0,1, then 101, then 1001. So what is a good upper limit for guessing? Aborting the code can be a safe option.

**Comment by prckent** [comment]
The general issue is how to make restarts super easy and the workflow much better than now.   There are several cases to consider.

If a run is guillotined due to walltime then restarts may not be possible at the moment in any case - that is a separate problem that could be tackled by rotating output files. We do have a maxcpu time setting to help avoid this. @aannabe is asking about a more common case where runs complete OK but statistics show a longer run is needed. This is the case that someone running on a workstation or local cluster is likely to encounter most.

Checking which sequences already exist would be trivial using the C++17 library filesystem capabilities, or simply trying to open name.sequenceNNN.scalar.dat before writing over it.

**Comment by aannabe** [comment]
Personally, the most convenient would be as follows. Let's say I have an input with VMC followed by DMC where walkers are saved every n blocks which also keeps last saved config as a backup (i.e we would have two different config files). Then re-running the same input would append the VMC and DMC runs by reading the corresponding config files. This should work regardless of how the job terminates since we have a backup of config file from previous save in case the current config file becomes corrupt due to limited wall time.

**Comment by aannabe** [comment]
From the discussion above, a proper restart is left to the responsibility of the user. In that regard, there is no issue here. The user needs to make sure that a different series name is used to avoid overwriting the files.

**Comment by prckent** [comment]
Thanks for revisiting this issue and closing it Gani. I still agree with your point + the earlier discussion - there needs to be a better way to restart and reduce error bars. It would be very user friendly. We will revisit this in future. However for now we have bigger fish to fry making the batched code fully robust and correct, so it remains the responsibility of the user to take care with file names. Every N blocks checkpointing was just reeenabled #4646 . If anyone is truly stuck with the current capabilities we will revisit sooner.
